src/Evolution/SlopeLimiters/Minmod.hpp

```diff
@@ -0,0 +1,413 @@
+  ///
+  /// \param type The type of Minmod limiter.
+  /// \param tvbm_constant The value of the TVBM constant (default: 0).
+  explicit Minmod(const MinmodType type,
+                  const double tvbm_constant = 0.0) noexcept
+      : type_(type), tvbm_constant_(tvbm_constant) {
+    ASSERT(tvbm_constant >= 0.0, "The TVBM constant must be non-negative.");
+  }
+
+  Minmod() noexcept = default;
+  Minmod(const Minmod& /*rhs*/) = default;
+  Minmod& operator=(const Minmod& /*rhs*/) = default;
+  Minmod(Minmod&& /*rhs*/) noexcept = default;
+  Minmod& operator=(Minmod&& /*rhs*/) noexcept = default;
+  ~Minmod() = default;
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& p) noexcept {  // NOLINT
+    p | type_;
+    p | tvbm_constant_;
+  }
+
+  const MinmodType& type() const noexcept { return type_; }
+  const double& tvbm_constant() const noexcept { return tvbm_constant_; }
+
+  /// \brief Computes data that must be communicated to neighbor elements.
+  ///
+  /// The minmod limiter needs only the cell-averaged means of the tensors in
+  /// each neighboring DG element. This function computes and packages the cell-
+  /// averaged data.
+  ///
+  /// \param means The cell-averaged means of each tensor.
+  /// \param args The tensors to be averaged and packaged.
+  /// \param mesh The mesh on which the tensor values are measured.
+  void data_for_neighbors(
+      const gsl::not_null<std::add_pointer_t<
+          Minmod_detail::tensor_double_from<db::item_type<Tags>>>>... means,
+      const db::item_type<Tags>&... args, const Mesh<VolumeDim>& mesh) const
+      noexcept {
+    const auto wrap_compute_mean = [&mesh](const auto& mean,
+                                           const auto& arg) noexcept {
+      for (size_t i = 0; i < arg.size(); ++i) {
+        (*mean)[i] = mean_value(arg[i], mesh);
+      }
+      return '0';
+    };
+    expand_pack(wrap_compute_mean(means, args)...);
+  }
+
+  /// \brief Limits the solution on the element.
+  ///
+  /// For each component of each tensor, the limiter will (in general) linearize
+  /// the data, then possibly reduce its slope, dimension-by-dimension, until it
+  /// no longer looks oscillatory.
+  ///
+  /// \param local_args The tensors to be limited.
+  /// \param neighbor_args The tensor cell-averages from each neighbor.
+  /// \param element The element on which the tensors to limit live.
+  /// \param mesh The mesh on which the tensor values are measured.
+  /// \param logical_coords The logical coordinates of the mesh gridpoints.
+  /// \param element_size The size of the element, in the inertial coordinates.
+  /// \param neighbor_sizes The sizes of the neighboring elements.
+  ///
+  /// \return whether the limiter modified the solution or not.
+  ///
+  /// \note The return value is false if the limiter knows it has not modified
+  /// the solution. True return values can indicate:
+  /// - The solution was limited to reduce the slope, whether by a large factor
```

Comment:
Another variation of this suggestion would be to fine-tune the current algorithm a bit so that the returned bool isn't set to `true` when the limiter only touches the solution by order roundoff.

I can see either of these being useful, but I think we should get this merged as is and only put in the work if/when we decide we want the feature.

Reply:
Sounds good!